awx/ui_next/src/components/Lookup/Lookup.test.jsx

```diff
@@ -1,285 +1,359 @@
 /* eslint-disable react/jsx-pascal-case */
 import React from 'react';
 import { createMemoryHistory } from 'history';
-import { mountWithContexts } from '../../../testUtils/enzymeHelpers';
+import { mountWithContexts, waitForElement } from '@testUtils/enzymeHelpers';
 import Lookup, { _Lookup } from './Lookup';
 
 let mockData = [{ name: 'foo', id: 1, isChecked: false }];
 const mockColumns = [{ name: 'Name', key: 'name', isSortable: true }];
-describe('<Lookup />', () => {
-  test('initially renders succesfully', () => {
-    mountWithContexts(
-      <Lookup
-        lookupHeader="Foo Bar"
-        name="fooBar"
-        value={mockData}
-        onLookupSave={() => {}}
-        getItems={() => {}}
-        columns={mockColumns}
-        sortedColumnKey="name"
-        multiple
-      />
-    );
-  });
 
-  test('API response is formatted properly', done => {
-    const wrapper = mountWithContexts(
-      <Lookup
-        lookupHeader="Foo Bar"
-        name="fooBar"
-        value={mockData}
-        onLookupSave={() => {}}
-        getItems={() => ({
-          data: { results: [{ name: 'test instance', id: 1 }] },
-        })}
-        columns={mockColumns}
-        sortedColumnKey="name"
-        multiple
-      />
-    ).find('Lookup');
+/**
+ * Check if an element exists on the document body
+ * @param {selector} query selector
+ * @param {expected} bool to assert existence or non-existence
+ */
+function checkRoot(selector, expected) {
+  const queryResult = global.document.querySelector(selector);
+  if (expected) {
+    expect(queryResult).not.toEqual(null);
+  } else {
+    expect(queryResult).toEqual(null);
+  }
+}
 
-    setImmediate(() => {
-      expect(wrapper.state().results).toEqual([
-        { id: 1, name: 'test instance' },
-      ]);
-      done();
-    });
+/**
+ * Check the lookup input group tags for expected values
+ * @param {wrapper} enzyme wrapper instance
+ * @param {expected} array of expected tag values
+ */
+async function checkInputGroup(wrapper, expected) {
```

Comment:
These helper functions are 💯! The names are just a little vague, if you have any ideas how to make it a little clearer what they assert without reading the docblock.

Reply:
agreed - I pushed up some updates to improve them